python/res/fm/ecl/ecl_config.py

```diff
@@ -0,0 +1,149 @@
+import os
+import yaml
+import sys
+
+
+def _replace_env_string(value):
+    while True:
+        dollar_pos = value.find("$")
+        if dollar_pos < 0:
+            break
+
+        end_pos = value.find(":", dollar_pos)
+        if end_pos < 0:
+            end_pos = len(value)
+
+        env_var = value[dollar_pos+1:end_pos]
+        env_value = os.getenv(env_var,"")
+        value = value.replace("${}".format(env_var), env_value)
+
+    return value
+
+
+# Small utility function will take a dict as input, and create a new dictionary
+# where all $VARIABLE in the values has been replaced with getenv("VARIABLE").
+
+def _replace_env(env):
+    new_env = {}
+    for key,value in env.items():
+        if "$" in value:
+            new_env[key] = _replace_env_string(value)
+        else:
+            new_env[key] = value
+
+    return new_env
+
+
+class Simulator(object):
+    """Small 'struct' with the config information for one simulator.
+    """
+    def __init__(self, name, version, executable, env, mpirun = None):
+        self.name = name
+        self.version = version
+        if not os.access( executable , os.X_OK ):
+            raise OSError("The executable: '{}' can not be executed by user".format(executable))
+
+        self.executable = executable
+        self.env = env
+
+        if mpirun is None:
+            self.mpi = False
+        else:
+            self.mpi = True
+            if not os.access(mpirun, os.X_OK):
+                raise OSError("The mpirun argument: '{}' is not executable by user".format(executable))
+
+        self.mpirun = mpirun
```

Comment:
`self.mpi` is a bit redundant, checking `self.mpirun is not None` is equally effective

Reply:
Removed